iphone/Maps/Core/DeepLink/DeepLinkHandler.swift

```diff
@@ -53,21 +57,27 @@
     LOG(.error, "handleDeepLink is called with nil URL")
     return false
   }
-  
-  private func handleDeepLink(url: URL) -> Bool {
-    LOG(.info, "handleDeepLink: \(url)")
 
-    if url.scheme == "file" {
-      // Import bookmarks.
-      DeepLinkParser.addBookmarksFile(url)
-      return true  // Async parsing can fail later, but here we always return true.
+  private func handleFileImport(url: URL) -> Bool {
+    LOG(.info, "handleFileImport: \(url)")
+    let fileCoordinator = NSFileCoordinator()
+    var error: NSError?
+    fileCoordinator.coordinate(readingItemAt: url, options: [], error: &error) { fileURL in
```

Comment:
It would be great to have a comment here that synchronous file reading is important to avoid permissions issue.